lab-taylor/controller/team-controller.js

```diff
@@ -0,0 +1,64 @@
+'use strict';
+
+const debug = require('debug')('authlab:team-controller');
+const httpErrors = require('http-errors');
+const Team = require('../model/team');
+
+exports.createTeam = function(teamData) {
+  debug('creating team');
+  return new Promise((resolve, reject) => {
+    new Team(teamData).save()
+    .then(team => resolve(team))
+    .catch(err => reject(httpErrors(400, err.message)));
+  });
+};
+
+exports.fetchTeam = function(teamId) {
+  debug('fetching teams', teamId);
+  return new Promise((resolve, reject) => {
+    Team.findOne({_id: teamId})
```

Comment:
You can cut down on code by writing:

Team.findById(teamId)
.then(resolve)
.catch( // code );